### PR TITLE
Break out tcf URLs and dataRetention into their own codecs

### PR DESCRIPTION
**Pull request by linh-transcend**
## Related Issues

- _[none]_

## Security Implications

_[none]_

## System Availability

_[none]_

### CODE REVIEW

File: src/iab.ts [hunk 1]
```diff
@@ -125,7 +125,42 @@ export const TcfGvlV3DataCategory = t.type({
...
+ * TCF GVL v3 URL configuration
+ */
+export const TcfGvlV3Urls = t.array(
+  t.intersection([
+    t.type({ langId: t.string, privacy: t.string }),
+    t.partial({
+      legIntClaim: t.string,
+    }),
+  ]),
+);
```
Comment (MadDataScience) [inline src/iab.ts:137]: I find 
```suggestion
export const TcfGvlV3Url = t.intersection([
  t.type({ langId: t.string, privacy: t.string }),
  t.partial({
    legIntClaim: t.string,
  }),
]);
```
a little more flexible. It's easier to wrap a type in `t.array` than to unravel it if you need.

File: src/iab.ts [hunk 2]
```diff
@@ -141,23 +176,8 @@ export const TcfGvlV3Vendor = t.intersection([
...
-    urls: t.array(
-      t.intersection([
-        t.type({ langId: t.string, privacy: t.string }),
-        t.partial({
-          legIntClaim: t.string,
-        }),
-      ]),
-    ),
+    dataRetention: TcfGvlV3DataRetention,
+    urls: TcfGvlV3Urls,
```
Comment (MadDataScience) [inline src/iab.ts:180]:
```suggestion
    urls: t.array(TcfGvlV3Url),
```
following the above suggestion.

**Review by MadDataScience — DISMISSED** [review]
a minor suggestion. (Then in `createTCFVendors.ts` you could put 
```typescript
urls: decodeCodec(t.array(TcfGvlV3Url), urls);
``` 
for instance.

File: src/iab.ts (hunk 2 above)
Comment (linh-transcend) [reply src/iab.ts]: Ahhh it much more flexible now! I made the change and it should be ready for re-review! Thanks!

**Review by MadDataScience — APPROVED** [review]
👌🏼

**Pull request merged**